eslint.config.mjs

```diff
@@ -0,0 +1,305 @@
+import js from '@eslint/js';
+import tseslint from '@typescript-eslint/eslint-plugin';
+import tsparser from '@typescript-eslint/parser';
+import react from 'eslint-plugin-react';
+import reactHooks from 'eslint-plugin-react-hooks';
+import cypress from 'eslint-plugin-cypress';
+import eslintComments from 'eslint-plugin-eslint-comments';
+import storybook from 'eslint-plugin-storybook';
+import compat from 'eslint-plugin-compat';
+import posthog from 'eslint-plugin-posthog';
+import simpleImportSort from 'eslint-plugin-simple-import-sort';
+import importPlugin from 'eslint-plugin-import';
+import unusedImports from 'eslint-plugin-unused-imports';
+import reactGoogleTranslate from 'eslint-plugin-react-google-translate';
+import jest from 'eslint-plugin-jest';
+import globals from 'globals';
+
+export default [
+    // Global ignores
+    {
+        ignores: [
+            'node_modules/**',
+            '**/node_modules/**',
+            'plugin-server/**',
+            'rust/**',
+            'livestream/**',
+            'common/hogvm/typescript/**',
+            'common/plugin_transpiler/**',
+            'common/hogvm/__tests__/**/__snapshots__/**',
+            'cypress/**/*.ts',
+            '.eslintrc.js',
+            'jest.config.ts',
+            'common/hogvm/__tests__/__snapshots__/**',
+            'common/hogvm/typescript/dist/**',
+            'common/hogvm/typescript/jest.config.js',
+            'common/plugin_transpiler/dist/**',
+            'common/plugin_transpiler/build.mjs',
+            '**/jest.config.js',
+            '**/*LogicType.ts',
+            '**/*.d.ts',
+            '**/__snapshots__/*.snap',
+            '.cursor/**',
+            '.dagster/**',
+            '.devcontainer/**',
+            '.flox/**',
+            '.ruff_cache/**',
+            '.vscode/**',
+            '.cache/**',
+            'dist/**',
+            'build/**',
+            'dags/**',
+            'funnel-udf/**',
+            'patches/**',
+            '**/__pycache__/**',
+            '**/*.pyc',
+            '**/*.pyo',
+            '**/*.pyd',
+            '**/*.py',
+        ]
+    },
+
+    // Base JavaScript config
+    js.configs.recommended,
+
+    // Main configuration
+    {
+        files: [
+            'frontend/**/*.{js,jsx,ts,tsx}',
+            'ee/frontend/**/*.{js,jsx,ts,tsx}',
+            'products/**/frontend/**/*.{js,jsx,ts,tsx}'
```

Comment:
added this to try and override behaviour

but products folders with a package.json

are resetting the config

and not getting eslinted 🙈